### DISCUSSION TITLE
Tab behaviour on comparisons of files vs folders

### DISCUSSION THREAD

**Discussion by boar24** [WinMerge/winmerge#1366] [Q&A] [answered]
Imagine my comparison result is like this:
![image](https://user-images.githubusercontent.com/52359342/173376689-9e53f01e-e685-4102-9ede-977a93b481db.png)
From the naming, I assume that the files Hartingplatte.JPG and Harting.JPG might be identical, but to make sure, I do a non-horizontal comparison by selecting the menu item as shown in the screenshot:
1) The non-horizontal file comparison of the two files opens in a new tab, which gets focus, so that I can see the comparison result.
2) After checking, I close the tab and are back in the previous tab.
3) I can continue to compare the next similarly named files, and so on.

**That's great!**

Now, imagine we have to do with folders, rather than with files. My comparison result be like this:
![image](https://user-images.githubusercontent.com/52359342/173375672-6383af26-7aca-47e6-91e1-c7fdfbaa69d7.png)
From the naming, I assume that the Folders Docs and Documents might have the same contents, but to make sure, I do a non-horizontal folder comparison by selecting the menu item as shown in the screenshot:
1) The non-horizontal folder comparison opens in the same tab. I can see the comparison result.
2) After checking, I want to continue with the next similarly named folders, but cannot easily. Rather, I will have to select the two parent folders again and repeat the comparison for these.

**That's not comfortable.**

A) Is there a trick to make the folder comparison open in a new tab that gets focus, so that the current tab is preserved?
B) Is there a reason, why in these cases the comparison of folders behaves differently than that of files?
C) Do you agree that the behaviours should in both cases be like for files?

**Comment by sdottaka** [comment, marked as answer]
> A) Is there a trick to make the folder comparison open in a new tab that gets focus, so that the current tab is preserved?

Holding down the Ctrl key and clicking on the menu item works as you would expect.

> B) Is there a reason, why in these cases the comparison of folders behaves differently than that of files?

This is because we want WinMerge to behave in the same way as when double-clicking a subfolder in Explorer.

However, in recent versions, except for the "Compare Non-horizontally" menu, I've changed the behavior to compare in a new tab only when 　enabling 'Include Subfolders' option.

The "Compare Non-horizontally" menu doesn't behave that way simply because I forgot to change it.

> C) Do you agree that the behaviours should in both cases be like for files?

I will fix it so that comparisons are made on new tabs only if the 'Include Subfolders' option is enabled.

**Reply by boar24** [reply]
A) Oh, cool, that works!
B) Ah, got it. 
C) Thanks a lot!

**Comment by sdottaka** [comment]
Fixed in commit 984f068
Thank you for the report.
